internal/logger/default_logger.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2024 Canonical.
+package logger
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/juju/zaputil/zapctx"
+	"github.com/mattn/go-colorable"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// SetupLogger sets up the default logger.
+// The local logger is a colorized plain text logger.
+// The production logger is a JSON structured logger.
+func SetupLogger(ctx context.Context, logLevel string, logLocal bool) {
+	pLogLevel, err := zapcore.ParseLevel(logLevel)
+	if err != nil {
+		fmt.Printf("ERROR: log level '%s' cannot be parsed, defaulting to info\n", logLevel)
+		pLogLevel = zap.InfoLevel
+	}
+	if logLocal {
+		devConfig := zap.NewDevelopmentEncoderConfig()
```

Comment:
Can we remove the tabs before hand? You can ask AI for build a pretty logging format for you also...

in internal/testutils/jimmtest/logging.go

we have a test logger, 

```go
func (s *LoggingSuite) setUp(c *gc.C) {
	output := gocheckZapWriter{c}
	logger := zap.New(zapcore.NewCore(
		zapcore.NewConsoleEncoder(zapcore.EncoderConfig{
			LevelKey:    "level",
			MessageKey:  "msg",
			EncodeLevel: zapcore.CapitalLevelEncoder,
			EncodeTime:  zapcore.ISO8601TimeEncoder,
		}),
		output,
		zap.DebugLevel,
	))

	zapctx.Default = logger

```

Ideally, they'd share the same development logger so it's consistent. Let's revisit the test part in this PR, I see in jimmsuite you're doing it on our suites setuptest, but then we're just setting two loggers up for no reason. So move the development config (with the amendments) to a centralised location that can be shared between the two:
1. The running server in dev mode
2. Our LoggingSuite updated with the same dev log config